wagtail/admin/views/generic/models.py

```diff
@@ -294,12 +296,11 @@ def search_queryset(self, queryset):
                 return search_backend.search(
                     self.search_query, queryset, fields=self.search_fields
                 )
+        query = Q()
+        for field in self.search_fields or []:
+            query |= Q(**{field + "__icontains": self.search_query})
 
-        filters = {
-            field + "__icontains": self.search_query
-            for field in self.search_fields or []
-        }
-        return queryset.filter(**filters)
+        return queryset.filter(filters)
```

Comment:
`filters` is not defined, maybe `return queryset.filter(query)`?

See https://github.com/wagtail/wagtail/actions/runs/6422261474/job/17445008042?pr=11000

This test output is also a good indication of what tests hit your code. I'd expect these tests to be updated/extended to also test your `__icontains` scenario.

Reply:
Sorry, that was a typo. I fixed it now. Thanks for pointing out.